src/NuGet.Licenses/Services/LicenseFileService.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Web;
+namespace NuGet.Licenses
+{
+    public class LicenseFileService : ILicenseFileService
+    {
+        public string GetLicenseFilePath(string licenseIdentifier)
+        {
+            return Path.GetFullPath(Path.Combine(HttpContext.Current.Server.MapPath("~\\App_Data\\Licenses\\"), String.Concat(licenseIdentifier, ".txt")));
+        }
+
+        public bool IsLicenseFilePathAllowed(string licenseFilePath)
+        {
+            return licenseFilePath.StartsWith(HttpContext.Current.Server.MapPath("~\\App_Data\\Licenses\\"));
+        }
+
+        public bool IsLicenseFileExisted(string licenseFilePath)
+        {
```

Comment:
We need to check `IsLicenseFilePathAllowed` here before checking if file exists.

Reply:
Done